packages/react-interactions/accessibility/src/FocusTable.js

```diff
@@ -69,30 +70,59 @@ function focusScope(cell: ReactScopeMethods, event?: KeyboardEvent): void {
   }
 }
 
-function focusCellByIndex(
+// This takes into account colSpan
+function focusCellByColumnIndex(
   row: ReactScopeMethods,
-  cellIndex: number,
+  columnIndex: number,
   event?: KeyboardEvent,
 ): void {
   const cells = row.getChildren();
   if (cells !== null) {
-    const cell = cells[cellIndex];
-    if (cell) {
-      focusScope(cell, event);
+    let colSize = 0;
+    for (let i = 0; i < cells.length; i++) {
+      const cell = cells[i];
+      if (cell) {
+        colSize += cell.getProps().colSpan || 1;
+        if (colSize > columnIndex) {
+          focusScope(cell, event);
+          return;
+        }
+      }
     }
   }
 }
 
+function getCellIndexes(
+  cells: Array<ReactScopeMethods>,
+  currentCell: ReactScopeMethods,
+): [number, number] {
+  let totalColSpan = 0;
+  for (let i = 0; i < cells.length; i++) {
+    const cell = cells[i];
+    if (cell === currentCell) {
+      return [i, i + totalColSpan];
+    }
+    const colSpan = cell.getProps().colSpan;
+    if (colSpan) {
+      totalColSpan += colSpan - 1;
+    }
+  }
+  return [-1, -1];
+}
+
 function getRowCells(currentCell: ReactScopeMethods) {
   const row = currentCell.getParent();
   if (row !== null && row.getProps().type === 'row') {
     const cells = row.getChildren();
     if (cells !== null) {
-      const rowIndex = cells.indexOf(currentCell);
-      return [cells, rowIndex];
+      const [rowIndex, rowIndexWithColSpan] = getCellIndexes(
+        cells,
+        currentCell,
+      );
+      return [cells, rowIndex, rowIndexWithColSpan];
```

Comment:
Do we even need to return the `rowIndex` without the colSpan?

Reply:
Yeah, we need it for when we move left and right still.